harness/determined/exec/prep_container.py

```diff
@@ -111,6 +111,44 @@ def do_rendezvous_slurm(
     return det.RendezvousInfo(container_addrs=addrs, container_rank=rank)
 
 
+# On HPC, the "launcher" tells the Determined Master that the job is "Running"
+# as soon as the workload manager (e.g., Slurm, PBS, etc) starts running the job.
+# However, if the container is not already cached on the compute node, it will
+# first need to be pulled down from the Internet by Singularity or Podman. From
+# the Determined Master's point of view, the experiment is not running until all
+# the containers are pulled and each container's entry point is executed. The
+# Determined Master will show a state of "Pulling" until it receives notification
+# from all the containers that they are running.  Therefore, notify the
+# Determined Master that the container is running, so that once all the
+# containers that are part of the job report they are running, the Determined
+# Master can change the state from "Pulling" to "Running".
+def send_container_running_notification(sess: api.Session, allocation_id: str) -> None:
```

Comment:
This is an HPC-only method and can't be called on other platforms.   Would be better if the method name included hpc_ perhaps and the comment was more explicit about only working for HPC launcher jobs.

Reply:
I can do that, but I'd prefer not to make any more changes at this point.  If Bradley requests changes, then I can certainly include your request.